### ISSUE TITLE
Feedback homework week 2

### ISSUE THREAD

**Issue by remarcmij** [Sob7i/hyf-javascript3#2]
Hi Sobhi, here is my feedback on your homework for week 2.

I'm not sure whether you actually read my feedback (in an ISSUE) that I gave you for week 1, because I don't see much of it being followed up in week 2. For instance, points 3, 4 and 6 of that feedback are not followed-up. Please do so for week 3.

**1.** You have embedded function `theListener()` inside function `renderData()`, which itself is embedded in function `main()`:

```js
function main() {
  // ...
  function renderData(data) {
    // ...
    function theListener(event) {
      // ...
    }
  }
}
```

There is no valid reason for doing it this way, and therefore it should not be done. Just put them all at the 'global' level, like this:

```js
function main() {
  // ...
}

function renderData(data) {
  // ...
}

function theListener(event) {
  // ...
}
```

**2.** I see lots of ESLint warnings (green underlines) in your `scripts.js` file, for instance about unused variables. Please resolve those issues.

**3.** Looking at this code snippet from your function `main()` (notice that I removed some unnecessary and meaningless blank lines):

```js
fetchJSON(motherUrl)
  .then(data => {
    renderData(data);
  })
  .catch(error => {
    console.error(error.message);
  });
```

In the `.then()` method you are passing an arrow function as its argument. This arrow function takes the data returned by `fetchJSON()` and then calls `renderData()` with that same data. You can therefore rewrite the code like this:

```js
fetchJSON(motherUrl)
  .then(renderData)
  .catch(error => {
    console.error(error.message);
  });
```

**4.** You did not implement rendering network errors to the HTML page rather than writing them to the console, as I asked in this [slack message](https://hackyourfuture.slack.com/files/U383PTTK9/FA1C6Q7L4/404.png).

**5.** Pay attention to how you name variables. Avoid words like `data` if you know the variable to contain a specific type of data. For instance, if you know that the data is about 'contributors', name your variable `contributors` and not `data`. Always choose the most descriptive name possible. The computer doesn't mind, but we, fellow humans reading your code _do_ mind.

**6.** I mentioned the use of blank lines. Don't add blank lines without thinking wether they add to the readability of your code. Use blank lines as if you were writing a piece of text. You would only use blank lines to separate paragraphs. Imagine reading a book where blank lines are inserted randomly. It wouldn't be a nice read.

If you want to have a look how I use blank lines, check out the example Nobel Prize app code [here](https://github.com/HackYourFuture/JavaScript3_examples/blob/master/week1/9-final/app.js).

**Bottom line**: code is working, promises are implemented but there are outstanding issues from week 1 that need to be taken care of.
